### ISSUE TITLE
Fixed: telemetry_cron should wait for the interval rather than running with every cron run.

### ISSUE THREAD

**Issue by yorkshire-pudding** [backdrop/backdrop-issues#6632]
## Description of the bug

While researching the telemetry module for the purposes of adding telemetry to a module, I noticed that no state was being set for telemetry and on closer inspection realised that the function will run whenever cron is invoked.

As state is never set, then `$last_run` will always be `FALSE`, which equates to `0`. `REQUEST_TIME - 0` will always be greater than `$interval`

<details>
<summary>telemetry_cron()</summary>

```php
/**
 * Implements hook_cron().
 */
function telemetry_cron() {
  if (settings_get('telemetry_enabled', TRUE)) {
    $last_run = state_get('telemetry_cron');
    $interval = 86400; // One day in seconds.
    if (REQUEST_TIME - $last_run > $interval) {
      $telemetry_data = telemetry_get_data();
      $destination_url = config_get('telemetry.settings', 'destination');
      $response = backdrop_http_request($destination_url, array(
        'method' => 'POST',
        'data' => backdrop_json_encode($telemetry_data),
        'headers' => array(
          'Content-Type' => 'application/json',
          'X-Site-Key' => _telemetry_site_key(),
        ),
      ));
    }
  }
}
````

</details>

In addition,  the module's `hook_uninstall` function is named incorrectly, but also uses a different state name of 'telemetry_cron_last'
I think this is a more appropriate state name given the other `cron_last` states.

## Steps To Reproduce

To reproduce the behavior:
1. Enable telemetry module
2. Run cron
3. Check database `state` table for anything related to telemetry

## Actual behavior
Nothing in state table for telemetry

## Expected behavior
A state in the state table for when telemetry cron was last run.

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version: 1.29.x-dev

**Comment by indigoxela** [comment]
The code looks good, but I wonder, how I can test this properly.

**Comment by argiepiano** [comment]
Thanks for catching this, @yorkshire-pudding. I've tested as follows:

Before the patch
- Ran cron
- Check the table `state`. There is no `telemetry_cron` or `telemetry_cron_last` entry. This indirectly proves that the call to the server is happening on every cron 👎🏽 

After the patch
- Ran cron
- Checked the table. `telemetry_cron_last` is there with a timestamp 👍🏽 
- Ran cron again. The value of that timestamp hasn't change 👍🏽 
- Uninstalled the telemetry mode. The `telemetry_cron_last` has been removed from `state` 👍🏽 

So, this LGTM
